Rustls 0.22 updates

On master, it seems that rustls version has been bumped but changes haven't been made. I get this when compiling:
```
error[E0432]: unresolved imports `rustls::Certificate`, `rustls::PrivateKey`
  --> src/tls_rustls/mod.rs:36:14
   |
36 | use rustls::{Certificate, PrivateKey, ServerConfig};
   |              ^^^^^^^^^^^  ^^^^^^^^^^ no `PrivateKey` in the root
   |              |
   |              no `Certificate` in the root
   |
   = help: consider importing one of these items instead:
           rustls::HandshakeType::Certificate
           rustls::internal::msgs::handshake::HandshakePayload::Certificate
error[E0277]: the trait bound `TlsAcceptor: From<Arc<rustls::ServerConfig>>` is not satisfied
  --> src/tls_rustls/future.rs:84:44
   |
84 | ...                   let acceptor = TlsAcceptor::from(server_config);
   |                                      ^^^^^^^^^^^ the trait `From<Arc<rustls::ServerConfig>>` is not implemented for `TlsAcceptor`
   |
   = help: the trait `From<Arc<tokio_rustls::rustls::ServerConfig>>` is implemented for `TlsAcceptor`
   = help: for that trait implementation, expected `tokio_rustls::rustls::ServerConfig`, found `rustls::ServerConfig`

error[E0599]: no method named `with_safe_defaults` found for struct `rustls::ConfigBuilder` in the current scope
   --> src/tls_rustls/mod.rs:273:10
    |
272 |       let mut config = ServerConfig::builder()
    |  ______________________-
273 | |         .with_safe_defaults()
    | |         -^^^^^^^^^^^^^^^^^^ method not found in `ConfigBuilder<ServerConfig, WantsVerifier>`
    | |_________|
    |

error[E0277]: the `?` operator can only be applied to values that implement `Try`
   --> src/tls_rustls/mod.rs:286:16
    |
286 |     let cert = rustls_pemfile::certs(&mut cert.as_ref())?;
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `impl Iterator<Item = Result<rustls_pki_types::CertificateDer<'static>, std::io::Error>> + '_`
    |
    = help: the trait `Try` is not implemented for `impl Iterator<Item = Result<rustls_pki_types::CertificateDer<'static>, std::io::Error>> + '_`

error[E0277]: the `?` operator can only be applied to values that implement `Try`
   --> src/tls_rustls/mod.rs:288:37
    |
288 |     let mut key_vec: Vec<Vec<u8>> = rustls_pemfile::read_all(&mut key.as_ref())?
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `impl Iterator<Item = Result<Item, std::io::Error>> + '_`
    |
    = help: the trait `Try` is not implemented for `impl Iterator<Item = Result<Item, std::io::Error>> + '_`

error[E0599]: no variant or associated item named `RSAKey` found for enum `Item` in the current scope
   --> src/tls_rustls/mod.rs:291:19
    |
291 |             Item::RSAKey(key) | Item::PKCS8Key(key) | Item::ECKey(key) => Some(key),
    |                   ^^^^^^ variant or associated item not found in `Item`

error[E0599]: no variant or associated item named `PKCS8Key` found for enum `Item` in the current scope
   --> src/tls_rustls/mod.rs:291:39
    |
291 |             Item::RSAKey(key) | Item::PKCS8Key(key) | Item::ECKey(key) => Some(key),
    |                                       ^^^^^^^^
    |                                       |
    |                                       variant or associated item not found in `Item`
    |                                       help: there is a variant with a similar name (notice the capitalization): `Pkcs8Key`

error[E0599]: no variant or associated item named `ECKey` found for enum `Item` in the current scope
   --> src/tls_rustls/mod.rs:291:61
    |
291 |             Item::RSAKey(key) | Item::PKCS8Key(key) | Item::ECKey(key) => Some(key),
    |                                                             ^^^^^ variant or associated item not found in `Item`

```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -41,7 +41,7 @@ tower = { version = "0.4", features = ["util"] }
 # optional dependencies
 ## rustls
 arc-swap = { version = "1", optional = true }
-rustls = { version = "0.22.0", optional = true }
+rustls = { version = "0.21", optional = true }
 rustls-pemfile = { version = "2.0.0", optional = true }
 tokio-rustls = { version = "0.24", optional = true }
 
@@ -51,7 +51,7 @@ tokio-openssl = { version = "0.6", optional = true }
 
 [dev-dependencies]
 serial_test = "2.0"
-axum = "0.7.1"
+axum = "0.7"
 hyper = { version = "1.0.1", features = ["full"] }
 tokio = { version = "1", features = ["full"] }
 tower = { version = "0.4", features = ["util"] }

--- a/src/tls_rustls/mod.rs
+++ b/src/tls_rustls/mod.rs
@@ -96,11 +96,11 @@ impl RustlsAcceptor {
         let inner = DefaultAcceptor::new();
 
         #[cfg(not(test))]
-        let handshake_timeout = Duration::from_secs(10);
+            let handshake_timeout = Duration::from_secs(10);
 
         // Don't force tests to wait too long.
         #[cfg(test)]
-        let handshake_timeout = Duration::from_secs(1);
+            let handshake_timeout = Duration::from_secs(1);
 
         Self {
             inner,
@@ -128,9 +128,9 @@ impl<A> RustlsAcceptor<A> {
 }
 
 impl<A, I, S> Accept<I, S> for RustlsAcceptor<A>
-where
-    A: Accept<I, S>,
-    A::Stream: AsyncRead + AsyncWrite + Unpin,
+    where
+        A: Accept<I, S>,
+        A::Stream: AsyncRead + AsyncWrite + Unpin,
 {
     type Stream = TlsStream<A::Stream>;
     type Service = A::Service;
@@ -283,12 +283,13 @@ fn config_from_der(cert: Vec<Vec<u8>>, key: Vec<u8>) -> io::Result<ServerConfig>
 fn config_from_pem(cert: Vec<u8>, key: Vec<u8>) -> io::Result<ServerConfig> {
     use rustls_pemfile::Item;
 
-    let cert = rustls_pemfile::certs(&mut cert.as_ref())?;
+    let cert = rustls_pemfile::certs(&mut cert.as_ref()).map(|it| it.map(|it| it.to_vec())).collect::<Result<Vec<_>, _>>()?;
     // Check the entire PEM file for the key in case it is not first section
-    let mut key_vec: Vec<Vec<u8>> = rustls_pemfile::read_all(&mut key.as_ref())?
-        .into_iter()
-        .filter_map(|i| match i {
-            Item::RSAKey(key) | Item::PKCS8Key(key) | Item::ECKey(key) => Some(key),
+    let mut key_vec: Vec<Vec<u8>> = rustls_pemfile::read_all(&mut key.as_ref())
+        .filter_map(|i| match i.ok()? {
+            Item::Sec1Key(key) => Some(key.secret_sec1_der().to_vec()),
+            Item::Pkcs1Key(key) => Some(key.secret_pkcs1_der().to_vec().into()),
+            Item::Pkcs8Key(key) => Some(key.secret_pkcs8_der().to_vec().into()),
             _ => None,
         })
         .collect();
@@ -376,8 +377,8 @@ mod tests {
             "examples/self-signed-certs/cert.pem",
             "examples/self-signed-certs/key.pem",
         )
-        .await
-        .unwrap();
+            .await
+            .unwrap();
 
         let server_handle = handle.clone();
         let rustls_config = config.clone();
@@ -505,7 +506,7 @@ mod tests {
                 "examples/self-signed-certs/cert.pem",
                 "examples/self-signed-certs/key.pem",
             )
-            .await?;
+                .await?;
 
             let addr = SocketAddr::from(([127, 0, 0, 1], 0));
 
@@ -565,7 +566,7 @@ mod tests {
                 _end_entity: &Certificate,
                 _intermediates: &[Certificate],
                 _server_name: &ServerName,
-                _scts: &mut dyn Iterator<Item = &[u8]>,
+                _scts: &mut dyn Iterator<Item=&[u8]>,
                 _ocsp_response: &[u8],
                 _now: SystemTime,
             ) -> Result<ServerCertVerified, rustls::Error> {

